src/hoverifier.ts

```diff
@@ -226,6 +241,9 @@ interface InternalHoverifierState<C extends object, D, A> {
     /** The currently hovered token */
     hoveredToken?: HoveredToken & C
 
+    /** The currently hovered token HTML element */
+    hoveredTokenElement?: HTMLElement
```

Comment:
The hover state is supposed to be props to be passed into a React component, so it seems weird to have an HTMLElement in it. The PR description doesn't contain any explanation for it, could you explain why this is needed?

There is also no guarantee that a "token" is its own HTML element. We actually want to move away from DOM mutation to wrap tokens in elements, so this would make it impossible to do that or the `hoveredTokenElement` would not only correspond to the hovered token but be an arbitrary element.

Another not nice thing is that the addition of this property allows for invalid state in this interface (`hoveredToken` can be set while `hoveredTokenElement` is not, where in reality that is an invalid state). This means it's easy to forget to reset one or the other in a call to `container.update()` and cause bugs.

Another minor concern is that currently all the properties in here are immune to the code host rerendering or mutating elements in the code view, but this element would maybe not be attached to the DOM anymore but then still be in the state and used by the consumer.

Reply:
The only reason to expose `hoveredTokenElement` is so that the Sourcegraph web app can set CSS properties and add a click listener to it https://github.com/sourcegraph/sourcegraph/pull/3444/files#diff-159ed2efd1fa050e21aa970011528ceeR224

I'm not very familiar with the codeintellify API. Is there a way to convert a `HoveredToken` into an HTML element? If so, this `hoveredTokenElement` field can be dropped. I think that would completely solve paragraphs 2 and 4.

If/when we move away from DOM mutation to wrap tokens in elements, couldn't we apply whatever technique we come up with for solving the problem of token highlighting to the problem of setting `cursor: pointer` and adding a click listener? I suppose it depends on the specific technique, but hypothetically if we overlay semi-transparent spans, I think it could work.

None of your concerns block merging of this PR, do they? This PR doesn't affect existing behavior. It would be great to be able to both experiment now AND improve the code asynchronously (best of both worlds).